src/rules/no-standalone-expect.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  AST_NODE_TYPES,
+  TSESTree,
+} from '@typescript-eslint/experimental-utils';
+import { createRule, isDescribe, isExpectCall, isFunction } from './tsUtils';
+
+const getBlockType = (stmt: TSESTree.BlockStatement) => {
+  const func = stmt.parent;
+  // functionDeclaration: function func() {}
+  if (func && func.type === AST_NODE_TYPES.FunctionDeclaration) {
+    return 'function';
+  } else if (func && isFunction(func) && func.parent) {
+    const expr = func.parent;
+    // arrowfunction or function expr
+    if (expr.type === AST_NODE_TYPES.VariableDeclarator) {
+      return 'function';
+      // if it's not a variable, it will be callExpr, we only care about describe
+    } else if (expr.type === 'CallExpression' && isDescribe(expr)) {
+      return 'describe';
+    }
+    return 'callExpr';
+  }
+  return false;
+};
+
+export default createRule({
+  name: __filename,
+  meta: {
+    docs: {
+      category: 'Best Practices',
+      description: 'Prevents expects that are outside of an it or test block.',
+      recommended: false,
+    },
+    messages: {
+      unexpectedExpect: 'Expect must be inside of a test block.',
+    },
+    type: 'suggestion',
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context) {
+    const callStack: String[] = [];
+
+    return {
+      CallExpression(node) {
+        if (isExpectCall(node)) {
+          const parent = callStack[callStack.length - 1];
+          if (!parent || parent === 'describe') {
+            context.report({ node, messageId: 'unexpectedExpect' });
+          }
+        }
+      },
+      BlockStatement(stmt) {
+        const blockType = getBlockType(stmt);
+        if (blockType) {
+          callStack.push(blockType);
+        }
+      },
+      'BlockStatement:exit'(stmt) {
```

Comment:
```suggestion
      'BlockStatement:exit'(stmt: TSESTree.BlockStatement) {
```

or is it correctly inferred? If it does, ignore me 🙂

Reply:
[It won't be inferred](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/experimental-utils/src/ts-eslint/Rule.ts#L221) as it matches `[nodeSelector: string]: RuleFunction | undefined;`.

Only properties that map directly to a `AST_NODE_TYPES` constant will be inferred.